### ISSUE TITLE
A way to disable the numbering in the playlist?

### ISSUE THREAD

**Issue by AhEhIhOhU** [clsid2/mpc-hc#1302]
Hi,
is there a way to disable numbering in the playlist?
The problem is that it gives a weird result when files with numbering in the name are loaded (and it happens all the time when you listen to an album).

Example:
![Example](https://user-images.githubusercontent.com/86187973/132884327-6428b0da-5d84-46fa-9018-f104332f7f91.png)

**Comment by adipose** [comment]
Maybe it could auto suppress leading digits if they match the sequence.

**Comment by clsid2** [comment]
You might have two albums in a single playlist, where only one has numbers.

I think most people will find that playlist numbers are useful information. So hiding them (optionally) just for cosmetic purposes isn't a great solution.

Maybe simply adjusting the layout of the numbering could make it look better.
Maybe `01. title` -> `01| title` or `[01] title`

But best solution would to create a M3U playlist yourself. Then you can customize the titles of the tracks.

**Comment by AhEhIhOhU** [comment]
> Maybe it could auto suppress leading digits if they match the sequence.


I'm not sure this is a good idea, some files have complex numbering
Example:
![Example - 00](https://user-images.githubusercontent.com/86187973/133803599-06fe2c4f-02eb-4bc2-b7e1-3766e6352f8a.png)

This can happen on an album from several CDs or on episodes of a series with several seasons. And as clsid2 said, just add 2 albums to break everything.


> But best solution would to create a M3U playlist yourself. Then you can customize the titles of the tracks.


Creating an M3U for each album would be extremely annoying, especially when you have hundreds of them...

> Maybe simply adjusting the layout of the numbering could make it look better.
> Maybe `01. title` -> `01| title` or `[01] title`


I did some tests:

![Example - 2](https://user-images.githubusercontent.com/86187973/133804158-227235f3-93f1-4c0b-b007-9bcd3781cebc.png)

![Example - 3](https://user-images.githubusercontent.com/86187973/133804193-5535373b-829d-48f7-b09a-9924241678fd.png)

(I don't know if it is technically possible to change the color or put a "hard" separation, but it can be interesting)

![Example - 4](https://user-images.githubusercontent.com/86187973/133804251-67e659b2-177e-4ff0-a902-4ddec0041aa2.png)

![Example - 5](https://user-images.githubusercontent.com/86187973/133804275-59ce9ecf-c38d-4c97-accd-0a1aca56c664.png)

Another idea that might be interesting is to put the information in the status bar as `[XX/YY]` and add an option to hide the numbering.
Since the status bar is completely empty when playing an audio file, this would add useful information that would also be visible when the playlist is hidden.

**Comment by clsid2** [comment]
One or two extra spaces between the number and title could maybe look better. The third image with the darker numbers looks nice.

@adipose, given that the playing item has different color, what do you think about just removing the selection dot in front?

**Comment by adipose** [comment]
Discussion about bullet here: https://github.com/clsid2/mpc-hc/issues/147

I believe it could be removed as I used it to identify the playing item rather than changing color (but then we did change the color).  However, the bullet does add a better visual for the currently playing item, compared to colors, particularly when the currently playing item is highlighted.  Although there's a specific color for the playing item, even when highlighted, there's nothing to contrast it to, so visually, it appears though it might "only" be highlighted as opposed to playing.  That's why I left it in originally.

Space between the numbers and the title seems like a good idea, though.  Although, the more I think about it, the more I think we don't need the numbers at all.  What functionality do they really add other than enumerating a sequence that's visually obvious?  Nobody is using the numbers for anything useful, I don't think, except MAYBE to figure out how many items are in the playlist when scrolling to the bottom?

**Comment by clsid2** [comment]
Playing + highlighted is visually different enough. So bullet can be removed.

Number can be useful for example to see position difference between two items. And to see how many items there are. Don't underestimate the amount of OCD some people have. But I agree that we could add an option to hide the numbers.

**Comment by adipose** [comment]
OK, I'll take a look at some point.  Maybe we can have two options:

1. Show bullet
2. Show numbers

Either way there's some space there between the current number or bullet and the playlist items.

**Comment by adipose** [comment]
![image](https://user-images.githubusercontent.com/3324395/134749292-523b3965-bf5f-43d0-87fb-4092a329ebc2.png)

![image](https://user-images.githubusercontent.com/3324395/134749301-9353a3d8-5ab7-49a2-a91c-21da7176f66e.png)

Here's some samples I made.  Bullet is gone, and I don't think it's needed anymore with the way I changed the text highlight.

**Comment by ale5000-git** [comment]
@adipose: I think it looks nice.

**Comment by AhEhIhOhU** [comment]
Effectively, the numbering that goes from gray to blue somehow replaces the ball.

With the darker color and the extra space, maybe the hooks are not necessary anymore ?

![test 3](https://user-images.githubusercontent.com/86187973/134769540-f541400d-e410-4cd4-8122-1e5f2014161c.png)

**Comment by AhEhIhOhU** [comment]
I take profit of this issue to ask a question: what is the utility of being able to select/highlight a file ?

In the Windows file explorer, it is useful because it is possible to select several files, which is impossible in the playlist (or I did not find how to do it).

I ask this question because I find it strange that a selected file is more visible than the file that is being read, which coupled with this behavior can be disturbing: #1313 
It may improve visibility when dragging and dropping or right-clicking, but isn't it better to deselect the file when the action is over? Or maybe when the playlist is not in focus ?

Maybe it has a functionality that I don't understand ?

**Comment by ale5000-git** [comment]
When you select an element you can press canc to remove it from the list, for example.

**Comment by clsid2** [comment]
Yes, the brackets are not really needed with the extra spacing and grey color.

I think using the same lighter blue color also for the highlighted case will look better.

**Comment by AhEhIhOhU** [comment]
> When you select an element you can press canc to remove it from the list, for example.

I asked the wrong question.
I should have asked instead: why does a file stay highlighted when it is not necessary ?
For example, when you press the space key to read the selected file (or double click on it), it moves the focus to the player, why does the file that is being read stay highlighted ?
It hinders visibility, in the same way as automatically selecting a file when it is open while the focus is not on the playlist.

To sum up, isn't it better to hide the highlighting when the focus is not on the playlist ?

PS: Is there a list of shortcuts for the playlist ?
A shortcut to switch the focus from the playlist to the player and back would be very useful.

PS 2: Isn't it more logical to use the enter key instead of the space key to read a file ?

**Comment by adipose** [comment]
Hiding (or fading) selection when not focused might be worth considering.

However, in general highlighting has a purpose: make it behave like all other list widgets (in windows and mpc). If the user clicked on an entry they did it for a reason and might want to know which one is currently selected.

Making it easy to clear selection might be an option, too.

**Comment by adipose** [comment]
As for brackets, I found it looked a bit better with them, as it nicely squares off, where numbers have different visual widths.

**Comment by clsid2** [comment]
Ok fine, then keep brackets.

One annoying thing that you can maybe also look at is the behavior when changing the name of a playlist entry. First issue is that it hides the numbering. But I don't know if it is easy to change this. Second is that it currently allows an empty title. It should just ignore the title change if new value is empty.

**Comment by adipose** [comment]
https://github.com/clsid2/mpc-hc/pull/1331

It was easy to prevent blank filenames.  The other thing is done, too.  It was a big funny to make work, due to that widget not rendering text in the default position for a listview.  It offsets text from the row rectangle instead of the designated text area (maybe a deliberate choice to make a bit more space for the playlist, not sure so I left it alone).  I added some DPI code to make it be a bit more predictable, and then was able to offset the inline edit so it overlays the original filename text.

**Comment by AhEhIhOhU** [comment]
Works perfectly.
Thanks a lot !

**Comment by Spuner** [comment]
It is impossible to turn off the numbering in the playlist, as I understand it?

**Comment by adipose** [comment]
Not currently.
